http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ResponseHandler.java

```diff
@@ -106,14 +109,52 @@ protected void channelRead0(ChannelHandlerContext channelContext, HttpObject msg
 
                 fullContent = ((FullHttpResponse) msg).content();
                 ByteBuffer bb = copyToByteBuffer(fullContent);
-                requestContext.handler().onStream(new FullResponseContentPublisher(channelContext, bb, ef));
-                finalizeResponse(requestContext, channelContext);
+                requestContext.handler().onStream(new DataCountingPublisher(channelContext,
+                                                                            new FullResponseContentPublisher(channelContext,
+                                                                                                             bb, ef)));
+
+                Optional<IOException> contentLengthFailure = validateResponseContentLength(channelContext);
```

Comment:
Nit: I think try/catch may be more conventional than `Optional` here.

Reply:
Changed